src/parameterized_tendencies/microphysics/precipitation.jl

```diff
@@ -163,16 +163,273 @@ end
 
 function compute_precipitation_cache!(Y, p, colidx, ::Microphysics1Moment, _)
     FT = Spaces.undertype(axes(Y.c))
-
+    (; params) = p
+    (; dt) = p.simulation
+    (; ᶜts) = p.precomputed
+    (; ᶜΦ) = p.core
     (; ᶜSqₜᵖ, ᶜSqᵣᵖ, ᶜSqₛᵖ, ᶜSeₜᵖ, ᶜwᵣ, ᶜwₛ) = p.precipitation
+    ᶜz = Fields.coordinate_field(Y.c).z
+
+    ᶜSᵖ = p.scratch.ᶜtemp_scalar
+
+    # get thermodynamics and 1-moment microphysics params
+    cmp = CAP.microphysics_params(params)
+    thp = CAP.thermodynamics_params(params)
+
+    # some helper functions to make the code more readable
+    Iₗ(ts) = TD.internal_energy_liquid(thp, ts)
+    Iᵢ(ts) = TD.internal_energy_ice(thp, ts)
+    qₗ(ts) = TD.PhasePartition(thp, ts).liq
+    qᵢ(ts) = TD.PhasePartition(thp, ts).ice
+    qᵥ(ts) = TD.vapor_specific_humidity(thp, ts)
+    Lf(ts) = TD.latent_heat_fusion(thp, ts)
+    Tₐ(ts) = TD.air_temperature(thp, ts)
+    #TODO
+    #cv_l(thp) = TD.Parameters.ThermodynamicsParameters.cv_l(thp)
```

Comment:
What is this TODO?

Reply:
I wanted to use derived parameters from Thermodynamics. Fixed now.